hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconContext.java

```diff
@@ -0,0 +1,140 @@
+ */
+@Singleton
+public final class ReconContext {
+  private static final Logger LOG = LoggerFactory.getLogger(ReconContext.class);
+
+  private final String threadNamePrefix;
+  private OzoneConfiguration ozoneConfiguration;
+  private ReconUtils reconUtils;
+  private boolean isHealthy = true;
+
+  private Map<ErrorCode, String> errCodeMsgMap = new HashMap<>();
+  private Map<ErrorCode, List<String>> errCodeImpactMap = new HashMap<>();
+  private List<ErrorCode> errors = new ArrayList<>();
+  private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+  @Inject
+  public ReconContext(OzoneConfiguration configuration, ReconUtils reconUtils) {
+    this.reconUtils = reconUtils;
+    this.ozoneConfiguration = configuration;
+    threadNamePrefix = reconUtils.getReconNodeDetails(configuration).threadNamePrefix();
+    initializeErrCodeMetaData();
+  }
+
+  private void initializeErrCodeMetaData() {
+    // This maps the error code to error message.
+    errCodeMsgMap.put(ErrorCode.OK, "Recon is healthy !!!");
+    errCodeMsgMap.put(ErrorCode.INVALID_NETWORK_TOPOLOGY,
+        "Invalid network topology of datanodes. Failed to register and load datanodes. Pipelines may not be " +
+            "healthy !!!");
+    errCodeMsgMap.put(ErrorCode.CERTIFICATE_INIT_FAILED, "Error during initializing Recon certificate !!!");
+    errCodeMsgMap.put(ErrorCode.INTERNAL_ERROR,
+        "Unexpected internal error. Kindly refer ozone-recon.log file for details !!!");
+    errCodeMsgMap.put(ErrorCode.GET_OM_DB_SNAPSHOT_FAILED, "OM DB Snapshot sync failed !!!");
+
+    // This maps the error codes to impacted areas of Recon.
+    errCodeImpactMap.put(ErrorCode.INVALID_NETWORK_TOPOLOGY, Arrays.asList("Datanodes", "Pipelines"));
+    errCodeImpactMap.put(ErrorCode.CERTIFICATE_INIT_FAILED, Arrays.asList("Initializing secure Recon"));
+    errCodeImpactMap.put(ErrorCode.INTERNAL_ERROR, Arrays.asList("Recon health"));
+    errCodeImpactMap.put(ErrorCode.GET_OM_DB_SNAPSHOT_FAILED, Arrays.asList("Overview (OM Data)", "OM DB Insights"));
+  }
+
+  /**
+   * @param healthStatus : update Health Status of Recon.
+   */
+  public void updateHealthStatus(boolean healthStatus) {
+    lock.writeLock().lock();
+    try {
+      LOG.info("Update healthStatus of Recon from {} to {}.", isHealthy, healthStatus);
+      this.isHealthy = healthStatus;
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public boolean isHealthy() {
+    lock.readLock().lock();
+    try {
+      return this.isHealthy;
+    } finally {
+      lock.readLock().unlock();
+    }
+  }
+
+  public String threadNamePrefix() {
+    return threadNamePrefix;
+  }
+
+  public Map<ErrorCode, String> getErrCodeMsgMap() {
+    return errCodeMsgMap;
+  }
+
+  public Map<ErrorCode, List<String>> getErrCodeImpactMap() {
+    return errCodeImpactMap;
+  }
+
+  public List<ErrorCode> getErrors() {
+    return errors;
```

Comment:
since errors list are protected for write, read also need return copy of list while return. 
Or we can use SynchronizedList, as well for isHealthy, can use AtomicBoolean and remove use of lock.